packages/wonder-blocks-form/src/components/text-field.tsx

```diff
@@ -159,100 +163,85 @@ type FullNumericInputProps = CommonProps & NumericInputProps;
 type Props = OtherInputProps | FullNumericInputProps;
 type PropsWithForwardRef = Props & WithForwardRef;
 
-type DefaultProps = {
-    type: PropsWithForwardRef["type"];
-    disabled: PropsWithForwardRef["disabled"];
-    light: PropsWithForwardRef["light"];
-};
-
-type State = {
-    /**
-     * Displayed when the validation fails.
-     */
-    errorMessage: string | null | undefined;
-};
-
 /**
  * A TextField is an element used to accept a single line of text from the user.
  */
-class TextField extends React.Component<PropsWithForwardRef, State> {
-    static defaultProps: DefaultProps = {
-        type: "text",
-        disabled: false,
-        light: false,
-    };
+const TextField = (props: PropsWithForwardRef) => {
+    const {
+        id,
+        type = "text",
+        value,
+        name,
+        disabled = false,
+        light = false,
+        error,
+        validate,
+        onValidate,
+        required,
+        placeholder,
+        style,
+        testId,
+        readOnly,
+        autoFocus,
+        autoComplete,
+        forwardedRef,
+        onKeyDown,
+        onChange,
+        onFocus,
+        onBlur,
+        // Should only include Aria related props
+        ...otherProps
+    } = props;
+    // Ensures error is updated on unmounted server-side renders
+    const [errorMessage, setErrorMessage] = React.useState(
+        (props.validate && props.value !== "" && props.validate(props.value)) ||
```

Comment:
This is something TextArea doesn't do currently, but can be something I add when we share validation logic between TextField and TextArea (if it makes sense to keep)! Currently TextArea [initializes the errorMessage to null](https://github.com/Khan/wonder-blocks/blob/e1c98df51ec152dbe80bd153d0e980caab436083/packages/wonder-blocks-form/src/components/text-area.tsx#L218)

For my own understanding: When I was converting this component, this logic is equivalent to setting the state errorMessage in the [constructor](https://github.com/Khan/wonder-blocks/pull/2356/files#diff-3f6913d896b8a5fb2855cf22cd787696ab51bb9833024d46344da4b4de3dbeccL187-L189), while the [useOnMountEffect](https://github.com/Khan/wonder-blocks/pull/2356/files#diff-3f6913d896b8a5fb2855cf22cd787696ab51bb9833024d46344da4b4de3dbeccR202-R204) logic is equivalent to the [componentDidMount](https://github.com/Khan/wonder-blocks/pull/2356/files#diff-3f6913d896b8a5fb2855cf22cd787696ab51bb9833024d46344da4b4de3dbeccL197-L201) logic. Are both needed? What is the use case for unmounted server-side renders?

Reply:
This was probably added for preventing hydration issues (SSR not being in sync with CSR). Here's more context about that: https://github.com/Khan/wonder-blocks/pull/1131#discussion_r643972638

Reply:
Note that we don't use SSR right now, but this would be turned on in the future, so maybe that's a feature we should consider?

Reply:
Thanks for the context! I will include it in the shared validation logic!